image.go

```diff
@@ -66,8 +67,18 @@ func (i *image) Unpack(ctx context.Context, snapshotterName string) error {
 	for _, layer := range layers {
 		unpacked, err := rootfs.ApplyLayer(ctx, layer, chain, sn, a)
 		if err != nil {
-			// TODO: possibly wait and retry if extraction of same chain id was in progress
-			return err
+			// wait and retry if extraction of same chain id was in progress
+			for retry := 3; retry > 0; retry-- {
+				// 3 second wait seems to be working for various diffrent image pull (parallel) tests.
+				time.Sleep(time.Second * 3)
```

Comment:
For validation, do you have tests against parllel image pulls with various sleeps ?

Reply:
I will add parallel tests.